left-sidebar: Add setting to display organization name in left sidebar.

anurastogiji opened the pull request:
Issue #20044 

We add setting to display organization name in left sidebar under `Personal settings > Display settings > Advanced` for user and `Organization > Default user settings > Advanced` for admin to control the default for new users.

![image](https://user-images.githubusercontent.com/75037620/150287935-06abf4f0-b936-4fd5-91f8-94ba4783f5b0.png)
![image](https://user-images.githubusercontent.com/75037620/150288511-6604e65a-d7cb-4e35-94e1-cd4efa932506.png)

![image](https://user-images.githubusercontent.com/75037620/150288066-4a2a2201-d235-4f89-974d-924f69dede46.png)
![image](https://user-images.githubusercontent.com/75037620/150288580-a6ef18d6-0c7d-4a44-92e7-d948f6cee8d0.png)

![image](https://user-images.githubusercontent.com/75037620/150288388-adf18581-ac75-412f-81de-5fa429899c7c.png)
![image](https://user-images.githubusercontent.com/75037620/150288694-d5b09338-fb69-4d02-b756-f98b66a354bc.png)

Documentation:
![image](https://user-images.githubusercontent.com/75037620/152781040-227ff554-7d0a-4b56-b841-38b50fbf7dbd.png)

timabbott commented on static/js/settings_config.ts:
```diff
@@ -492,6 +492,7 @@ export const display_settings_labels = {
             defaultMessage: "Convert emoticons before sending (<code>:)</code> becomes 😃)",
         }),
     ),
+    display_organization_name: $t({defaultMessage: "Display organization name"}),
```
This setting name needs to be more specific.

anurastogiji replied:
How about "Display organization name in left-sidebar"?

timabbott commented on static/templates/left_sidebar.hbs:
```diff
@@ -1,5 +1,10 @@
+        <div class= "company-name">
+            {{realm_name}}
+        </div>
+        {{/if}}
```
I think you want a persistent element that is hidden-or-not, so that we don't need to rerender the template to live update after the setting is changed.

Also, the name `company-name` isn't appropriate; since that's not what we call that in zulip and many users are not companies.

laurynmm commented on templates/zerver/api/changelog.md:
```diff
@@ -20,6 +20,12 @@ format used by the Zulip server that they are interacting with.
+
+* [`POST /register`](/api/register-queue), [`PATCH /settings`](/api/update-settings),
+  [`PATCH /realm/user_settings_defaults`](/api/update-realm-user-settings-defaults):
+  Added user setting `display_organization_name`.
```
I think adding a little description about what the setting does would be helpful here.

laurynmm commented on zerver/openapi/zulip.yaml:
```diff
@@ -7555,6 +7555,14 @@ paths:
+          description: |
+            Whether display the organization name in left-sidebar.
+            **Changes**: New in Zulip 5.0 (feature level 114).
+          schema:
```
Small typo here, you're missing 'to' after 'Whether'.

Also, adding an empty line between the setting description and the **Changes** note will make sure these updates are rendered correctly in the api documentation. Same note below for the other additions.

I find it helpful to always check any updates to this file in the api docs generated in the dev environment, which you can access by going to: localhost:9991/api/

laurynmm reviewed (commented):
@anurastogiji - I did a quick review of your PR and it looks good. I had a few small comments regarding the API documentation updates.

You might also want to start a conversation in the [CZO documentation stream](https://chat.zulip.org/#narrow/stream/19-documentation) about if and where it might be good to document this new setting for users in the help center.

Also, updating the PR with current screenshots of these changes since I think they've changed since the initial comment. 🙂

laurynmm reviewed (commented):
@anurastogiji - I did a quick review of the new help center documentation page just for a first pass and the tabs / links look good.

I was thinking maybe you could revise the first section text to describe what users can do to identify the organization. E.g. 'In the web app, you can identify the organization by ...' and 'In the Desktop app, the organization logo is ...' 

I think the page for '[Change your language](https://zulip.com/help/change-your-language)' could be a good reference for what I'm thinking. And it also has a small section directed to admins for the organization as well since this would be a useful feature for them to know about for new user settings.

Also, posting a screenshot of the new documentation page can be helpful for reviewing because you can see what the users will be reading.

anurastogiji commented:
I referred the page for '[Change your language](https://zulip.com/help/change-your-language)', made few changes and added documentation for admins too.

For new screenshots, I updated the first comment in this PR.

alya commented:
@anurastogiji could you please split off the help center page into a separate commit?

alya commented on templates/zerver/help/include/sidebar_index.md:
```diff
@@ -52,8 +52,10 @@
 * [Configure default view](/help/configure-default-view)
 * [Enable full width display](/help/enable-full-width-display)
 * [Display the buddy list on narrow screens](/help/move-the-users-list-to-the-left-sidebar)
+* [Display organization name in left-sidebar](/help/display-organization-name)
```
Let's use "Display organization name" for the title here, so that it's not overly long.

alya commented:
You can drop the display-organization-name-new-users page altogether, as it's covered by https://zulip.com/help/configure-default-new-user-settings. We don't need to document it separately.

alya commented:
@timabbott should we try it out on CZO and see how we like it? @laurynmm or I can do a polish pass on the documentation once we know we're happy with the feature.

anurastogiji committed:
left-sidebar: Add setting to display organization name in left sidebar.

We add setting to display organization name in left sidebar under
`Personal settings > Display settings > Advanced` for user and
`Organization > Default user settings > Advanced` for admin to control
the default for new users.

Fix #20044

anurastogiji committed:
documentation: Add documentation for display organization name.

Added section `Display organization name` under `Display settings`

anurastogiji commented:
> @anurastogiji could you please split off the help center page into a separate commit?

Done.

amanagr commented:
Did few minor edits:

```diff
diff --git a/static/styles/zulip.css b/static/styles/zulip.css
index a637f4fcce..961621edd3 100644
--- a/static/styles/zulip.css
+++ b/static/styles/zulip.css
@@ -429,7 +429,7 @@ p.n-margin {
             vertical-align: top;
             font-size: 1rem;
             font-weight: 600;
-            cursor: pointer;
+            cursor: text;
             max-width: $left_sidebar_width;
             padding-right: 10px;
             padding-left: 10px;
diff --git a/templates/zerver/help/include/sidebar_index.md b/templates/zerver/help/include/sidebar_index.md
index faeb1d813e..eb8812389a 100755
--- a/templates/zerver/help/include/sidebar_index.md
+++ b/templates/zerver/help/include/sidebar_index.md
@@ -55,7 +55,6 @@
 * [Display organization name](/help/display-organization-name)
 * [Manage your uploaded files](/help/manage-your-uploaded-files)
 
-
 ## Sending messages
 * [Open the compose box](/help/open-the-compose-box)
 * [Resize the compose box](/help/resize-the-compose-box)
 ```

LGTM!
